Automattic/wp-calypso
# Payment Methods: Use TaxFields for tax info dialog

#### Changes proposed in this Pull Request

When assigning a payment method to an existing subscription, a tax info dialog can be displayed to edit the tax location information (postal code and country) for a stored card; that dialog uses a bespoke form for its fields. However, there is an existing `TaxFields` component which can do this already.

This PR switches the dialog to use `TaxFields` instead. 

This also adds unit tests for the basic behaviors of the payment method.

<img width="742" alt="Screen Shot 2022-03-21 at 1 34 50 PM" src="https://user-images.githubusercontent.com/2036909/159331314-c1638629-5afd-426e-a927-9fa083aa6a2e.png">

<img width="572" alt="Screen Shot 2022-03-19 at 7 10 42 PM" src="https://user-images.githubusercontent.com/2036909/159141476-7cfda7dc-2d7e-43d3-9be7-7d4d766b7e62.png">

#### Testing instructions

You'll need to use an account with at least one active subscription and at least one stored card with no tax info. If you don't already have a card that's missing tax info, this can be tricky since we actively prevent you from adding one now (see https://github.com/Automattic/wp-calypso/pull/61862). The "easiest" way may be to manually delete tax info from a stored card on the server.

- Visit `/me/purchases` and click on an active subscription.
- Click on "Add payment method" or "Change payment method".
- Click on the "Missing required Country field" button on a payment method.
- Enter an _invalid_ country and postal code into the form and submit it.
- Verify that you see an error message about invalid data.
- Enter a valid country and postal code into the form and submit it.
- Verify that the country and postal code are added and now show up on the payment method in the list.
